src/Render/Program.cs

```diff
@@ -46,10 +47,13 @@ internal static class Program {
 			var text = await File.ReadAllTextAsync( filename );
 
 			var doc = MarkdownFactory.Parse( text );
-
 			doc.ApplyD2LTweaks();
-
-			MarkdownFactory.Render( outputHtml, doc );
+			var html = MarkdownFactory.RenderToString( doc );
+			var renderer = new TemplateRenderer( 
+				await File.ReadAllTextAsync( "Templates/page.html" )

```

Comment:
I tihnk you switched this to an embedded resource, so we could grab it from there via `ResourceManager` instead.

For stuff like this what can be nice is a static `Create` method on `TemplateRenderer` that does all that gross stuff. Think of it like `IFactory<T>` in the LMS. The idea is to keep the constructors simple, but have wrappers around them in static `Create` methods adjacent to the constructor. This is a common idiom in .NET.

Reply:
Should I hide the constructor, or expose both the constructor and `Create()`?

Reply:
I'm fine either way.

Reply:
Do you have an example of using ResourceManager for this? I can't find anything aside from `.resx` files and Reflection.

Reply:
Sorry I meant this buddy https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getmanifestresourcestream?view=netframework-4.8